src/main/java/frc/robot/commands/IntakeSequence.java

```diff
@@ -1,46 +1,68 @@
 package frc.robot.commands;
 
-import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import edu.wpi.first.wpilibj2.command.ParallelRaceGroup;
+import edu.wpi.first.wpilibj2.command.RunCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+import edu.wpi.first.wpilibj2.command.WaitUntilCommand;
+import frc.robot.Cal;
 import frc.robot.subsystems.Intake;
 import frc.robot.subsystems.Lift;
+import frc.robot.subsystems.Lift.LiftPosition;
 
-public class IntakeSequence extends CommandBase {
+public class IntakeSequence extends SequentialCommandGroup {
   private Intake intake;
   private Lift lift;
 
-  public IntakeSequence(Intake intake, Lift lift) {
-    this.intake = intake;
-    this.lift = lift;
-  }
+  // deploy intake for specified amount of time
+  private ParallelRaceGroup deployIntake =
+      new RunCommand(intake::deploy, intake).until(intake::atDesiredPosition);
 
-  @Override
-  public void initialize() {
-    intake.deploy();
-    lift.setDesiredPosition(Lift.LiftPosition.GRAB_FROM_INTAKE);
-  }
+  // run intake
+  private InstantCommand runIntake = new InstantCommand(intake::intakeGamePiece, intake);
 
-  @Override
-  public void execute() {
-    if (!lift.seeGamePiece()) {
-      intake.intakeGamePiece();
-    } else {
-      lift.closeGrabber();
-    }
-    // TODO fix this
-  }
+  // trigger the lift to move to the intake position. This does not need a timeout because it is
+  // running in the parallel group, which is controlled by lift.readyToIntake()
+  private InstantCommand liftToIntakePos =
+      new InstantCommand(() -> lift.setDesiredPosition(LiftPosition.GRAB_FROM_INTAKE), lift);
 
-  @Override
-  public boolean isFinished() {
-    return lift.holdingGamePiece() ? true : false;
-  }
+  // triggers the grabber to open when it is safe. Again, we do not care about time because it is in
+  // the parallel group.
+  private InstantCommand openGrabber = new InstantCommand(lift::openGrabber, lift);
+
+  // completes when the lift is at the correct position and the intake sees a game piece
+  private WaitUntilCommand waitUntilReadyToIntake =
+      new WaitUntilCommand(() -> (lift.atIntakePos() && intake.seeGamePiece()));
```

Comment:
Can you just call `atPosition`? It was just made public in another PR (you should rebase)